lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/NetworkConnectionFactory.java

```diff
@@ -73,16 +76,27 @@
     return connection;
   }
 
-  <T> Link<NetworkConnectionServiceMessage<T>> openLink(final Identifier remoteId) throws NetworkException {
+  Link<NetworkConnectionServiceMessage<T>> openLink(final Identifier remoteId) throws NetworkException {
     return networkService.openLink(remoteId);
   }
 
-  String getConnectionFactoryId() {
-    return this.connFactoryId;
+  @Override
+  public Identifier getConnectionFactoryId() {
+    return connectionFactoryId;
+  }
+
+  @Override
+  public Identifier getEndPointId() {
+    return endPointId;
   }
 
   Identifier getSrcId() {
-    return this.networkService.getNetworkConnectionServiceId();
+    // TODO : Change to always return the endPointId when the deprecated getNetworkConnectionServiceId is removed.
```

Comment:
This requires the remover to change (admittedly simple) code logic. 

A cleaner way to do this is to have a constructor without endPointId (that calls `this(..., this.networkService.getNetworkConnectionServiceId())`), and then mark that constructor as deprecated. This should be easier to notice and remove. Deprecating a non-public constructor might look awkward, but I think this will be preferred by whoever cleans up the code after 0.13.